### PR TITLE
Add storybook

### PR DESCRIPTION
**Pull request by lubej**
Add Storybook stories for:
- colors;
- typography;
- buttons;
- table;
- bar chart;
- line chart;

### CODE REVIEW

File: src/stories/LineChart.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { ComponentMeta, Story } from '@storybook/react'
+import Box from '@mui/material/Box'
+import React from 'react'
+import { LineChart } from '../app/components/charts/LineChart'
+import { intlDateFormat } from '../app/utils/dateFormatter'
+
+export default {
+  title: 'Example/LineChart',
```
Comment (buberdds) [inline src/stories/LineChart.stories.tsx:8]: Let's charts under storybook charts folder 
`title: 'Example/Charts/LineChart',` 
`title: 'Example/Charts/BarChart',`

File: src/stories/Buttons.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  size: {
+    control: 'select',
+    options: ['small', 'medium', 'large'] as unknown as ButtonTypeMap['props']['size'],
+  },
+}
+
+export const Secondary = Template.bind({})
+Secondary.args = {
+  children: 'Secondary',
+  color: 'primary',
```
Comment (buberdds) [inline src/stories/Buttons.stories.tsx:29]: secondary

File: src/stories/Buttons.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  size: {
+    control: 'select',
+    options: ['small', 'medium', 'large'] as unknown as ButtonTypeMap['props']['size'],
+  },
+}
+
+export const Tertiary = Template.bind({})
+Tertiary.args = {
+  children: 'Tertiary',
+  color: 'primary',
```
Comment (buberdds) [inline src/stories/Buttons.stories.tsx:42]: tertiary

File: .storybook/main.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  stories: ['../src/**/*.stories.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
+  addons: [
+    '@storybook/addon-links',
+    '@storybook/addon-essentials',
+    '@storybook/addon-interactions',
+    'storybook-addon-react-router-v6',
```
Comment (buberdds) [inline .storybook/main.js:7]: later we can look at adding @storybook/addon-a11y

File: package.json [hunk 5]
```diff
@@ -51,16 +56,27 @@
...
+    "@storybook/addon-interactions": "^6.5.15",
+    "@storybook/addon-links": "^6.5.15",
+    "@storybook/builder-webpack4": "^6.5.15",
+    "@storybook/manager-webpack4": "^6.5.15",
+    "@storybook/react": "^6.5.15",
+    "@storybook/testing-library": "^0.0.13",
     "@types/node": "^18.11.13",
     "@types/react": "^18.0.26",
     "@types/react-dom": "^18.0.9",
+    "babel-loader": "^8.3.0",
```
Comment (buberdds) [inline package.json:73]: Why we need this ? Doesn't storybook use ts-loader under the hood?

File: package.json [hunk 6]
```diff
@@ -51,16 +56,27 @@
...
   },
   "devDependencies": {
+    "@babel/core": "^7.20.7",
     "@parcel/packager-raw-url": "^2.8.1",
     "@parcel/transformer-webmanifest": "^2.8.1",
+    "@storybook/addon-actions": "^6.5.15",
+    "@storybook/addon-essentials": "^6.5.15",
+    "@storybook/addon-interactions": "^6.5.15",
+    "@storybook/addon-links": "^6.5.15",
+    "@storybook/builder-webpack4": "^6.5.15",
```
Comment (buberdds) [inline package.json:66]: we should go with webpack5

File: package.json (hunk 5 above)
Comment (lubej) [reply package.json:73]: It is actually a peer-dependency of webpack loader, so I guess not:

```
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "babel-loader"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "babel-loader@8.3.0"
info Has been hoisted to "babel-loader"
info Reasons this module exists
   - Specified in "devDependencies"
   - Hoisted from "@storybook#builder-webpack5#babel-loader"
   - Hoisted from "@storybook#manager-webpack5#babel-loader"

```

File: src/app/components/charts/BarChart.tsx [hunk 7]
```diff
@@ -10,12 +10,14 @@ import {
...
-export type BarChartProps = Formatters & {
-  data: any[]
-  dataKey: string
+interface BarChartProps<T extends object> extends Formatters {
+  data: T[]
+  dataKey: keyof T
 }
 
-export const BarChart: FC<BarChartProps> = ({ data, dataKey, formatters }) => (
+// const LineChartCmp = <T extends object>({
```
Comment (buberdds) [inline src/app/components/charts/BarChart.tsx:18]: leftover

**Review by buberdds — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (buberdds) [reply package.json:73]: ah so yeah I guess we don't need that in our package.json

File: src/app/components/charts/BarChart.tsx (hunk 7 above)
Comment (lubej) [reply src/app/components/charts/BarChart.tsx]: Good catch 👍

**Pull request merged**